### PR TITLE
Add clear methods for nodes and textures

### PR DESCRIPTION
**Pull request by griffin2000**
Filament had no way of clearing a texture on a material, or a parent on a instance.  This adds them.

Uses the convention _clearThing()_ rather than _setThing(NULL)_ to avoid errors in case NULL is not intentional.

### CODE REVIEW

**Review by an-kumar — COMMENTED** [review]
I don’t really get why this is necessary. Is it more than syntactic sugar Instead of set(null)? Are you saying filament doesn’t have some core functionality?

I really don’t think we should diverge too much from filament.

**Comment by an-kumar** [comment]
Yeah, as an example, https://github.com/google/filament/blob/d7e5f2c26b835454efbc55996aa0404be7a0a5a9/libs/gltfio/src/ResourceLoader.cpp#L572 is how to unset parents in the transform. simply setParent(instance, 0)... would it be possible to use that pattern?

similarly for the texture, could we expose the ability to create the null texture that you use in `setParameter`? Then we won't need to change filament... if at all possible it seems like we should try to stay close to filament's head

**Comment by griffin2000** [comment]
> Yeah, as an example, https://github.com/google/filament/blob/d7e5f2c26b835454efbc55996aa0404be7a0a5a9/libs/gltfio/src/ResourceLoader.cpp#L572 is how to unset parents in the transform. simply setParent(instance, 0)... would it be possible to use that pattern?
> 
> similarly for the texture, could we expose the ability to create the null texture that you use in `setParameter`? Then we won't need to change filament... if at all possible it seems like we should try to stay close to filament's head

Quite right.  The JS needs a clear method (as you can't pass a NULL object, and its error prone to do so)

There are no need for the C++ methods.

**Comment by griffin2000** [comment]
Though the clearParent is needed as that takes an instance object.  The clearCamera and clearScene are not required on the C++ side removed them.

**Review by an-kumar — COMMENTED** [review]
Why is clear parent needed? Instead of setparent(instance, 0)

File: web/filament-js/jsbindings.cpp [hunk 1]
```diff
@@ -780,7 +786,11 @@ class_<TransformManager>("TransformManager")
     .function("destroy", &TransformManager::destroy)
     .function("setParent", &TransformManager::setParent)
     .function("getParent", &TransformManager::getParent)
-
+    .function("clearParent", EMBIND_LAMBDA(void,
```
Comment (andresubiquity6) [inline web/filament-js/jsbindings.cpp:789]: i'm probably missing something but can we not just call: 'setParent' with '0'? and below: setTextureParameter with NULL?

**Review by andresubiquity6 — APPROVED** [review]
See comments please.

**Pull request merged**